### PR TITLE
Deprecate SbFile types.

### PR DESCRIPTION
**Pull request by yjzhang111**
b/302715109

### CODE REVIEW

File: starboard/shared/win32/file_internal.cc [hunk 1]
```diff
@@ -25,7 +25,7 @@
 #include "starboard/shared/win32/error_utils.h"
 #include "starboard/shared/win32/wchar_utils.h"
 
-#define O_ACCMODE O_RDWR | O_WRONLY | O_RDONLY
+#define O_ACCMODE (O_RDONLY | O_WRONLY | O_RDWR)
```
Comment (haozheng-cobalt) [inline starboard/shared/win32/file_internal.cc:28]: Does the flags order make any difference?

File: starboard/shared/win32/file_internal.cc [hunk 2]
```diff
@@ -110,6 +110,10 @@ HANDLE OpenFileOrDirectory(const char* path, int flags) {
   }
 
   DWORD creation_disposition = 0;
+  if (!flags) {
+    creation_disposition = OPEN_EXISTING;
```
Comment (haozheng-cobalt) [inline starboard/shared/win32/file_internal.cc:114]: why are we doing this?

File: starboard/shared/win32/file_internal.cc (hunk 1 above)
Comment (yjzhang111) [reply starboard/shared/win32/file_internal.cc:28]: No but I think the missing bracket was causing some issue in Windows. We don't ship SB17 windows anymore but it'd still be good to correct this.

File: starboard/shared/win32/file_internal.cc (hunk 2 above)
Comment (yjzhang111) [reply starboard/shared/win32/file_internal.cc:114]: This correspond to http://shortn/_xC9YiezPWn this section. It was missing in previously.

**Review by haozheng-cobalt — APPROVED** [review]

**Pull request merged**